packages/core/src/internal/sagaError.js

```diff
@@ -64,10 +69,14 @@ export const setCrashedEffect = effect => {
 export const toString = () => {
   const [firstSaga, ...otherSagas] = sagaStack
   const crashedEffectLocation = firstSaga.crashedEffect ? effectLocationAsString(firstSaga.crashedEffect) : null
-  const errorMessage = `The above error occurred in task ${sagaLocationAsString(firstSaga.meta)}${
+  let errorMessage = `The above error occurred in task ${sagaLocationAsString(firstSaga.meta)}${
     crashedEffectLocation ? ` \n when executing effect ${crashedEffectLocation}` : ''
   }`
 
+  if (process.env.NODE_ENV !== 'production' && !firstSaga.meta.location) {
```

Comment:
@restrry u know that logic better, is this condition a good one for this check? Can it yield false positives?

Reply:
unfortunately it doesn't work as expected. for errors raised in helpers we don't have `location`, as effect is `hidden` in library code. for example, one of such cases:
```js
  yield retry(3, 10, function() {
    undefinedIsNotAFunction()
  })
```
as an alternative solution we can use
```js
sagaStack.some(s => s.meta.location)
```
which is a bit better, but still could lead to false positives.

Reply:
hm, shouldnt "lastSaga" have `location` nearly in all cases then? 🤔

Reply:
I suppose so

Reply:
@Andarist do you want me to finish work on this branch?

Reply:
That would definitely help, I'll try to dig into the babel location stuff in the meantime